### PR TITLE
fix(specmap): limit depth of traversal

### PR DESCRIPTION
**Pull request by glowcloud**
Refs https://github.com/swagger-api/swagger-ui/issues/9513

Fixes the issue by limiting the depth of traversal but leaves some of the references unresolved.

### CODE REVIEW

File: test/specmap/complex.js [hunk 1]
```diff
@@ -49,4 +50,17 @@ describe('complex', () => {
...
+    // Given
+    const spec = globalThis.loadJsonFile(
+      path.join(__dirname, 'data', 'specs', 'complex-example.json')
+    );
+
+    // When
+    const result = await Swagger.resolve({ spec });
+
+    // Then
+    expect(result).not.toEqual(null);
```
Comment (glowcloud) [inline test/specmap/complex.js:64]: Not sure if this is fine or if perhaps we should check some schema here.

Comment (glowcloud) [reply test/specmap/complex.js:64]: Matching a snapshot here seems like a bad idea to me as the generated snapshot has a size over a 100 MB.

Comment (glowcloud) [reply test/specmap/complex.js:64]: Perhaps we could do a check for `$ref` and `$$ref` to show that the spec is only partially resolved? And the test name should also be changed to `'should partially resolve...'`

```
expect(
      result.spec.components.schemas[
        'com.sap.ctsm.backend.core.api.study.v1.StudyAPIv1.StudyTreatments-create'
      ].properties.scenario.allOf[0].$ref
    ).toEqual(
      '#/components/schemas/com.sap.ctsm.backend.core.api.study.v1.StudyAPIv1.Scenarios-create'
    );
```

 ```
   expect(
      result.spec.components.schemas[
        'com.sap.ctsm.backend.core.api.study.v1.StudyAPIv1.BlindingGroups'
      ].properties.study.properties.scenarios.items.$$ref
    ).toEqual('#/components/schemas/com.sap.ctsm.backend.core.api.study.v1.StudyAPIv1.Scenarios');
```

**Review by char0n — APPROVED** [review]
LGTM. Nice work!

**Pull request merged**